### ISSUE TITLE
incremental learning

### ISSUE THREAD

**Issue by julama** [smartcorelib/smartcore#46]
Hey, thanks for sharing this project!
Is incremental learning supported by smartcore; something like `partial_fit` from sk-learn?

**Comment by VolodymyrOrlov** [comment]
Hi Julian, this is a great direction to explore. I personally do not use `partial_fit` to train my models, but I can see where incremental learning might be useful. I know Scikit-learn provides only a handful of incremental estimators, which one are you interested in?

**Comment by julama** [comment]
The reason why I'm interested in `partial_fit` or mini-batches is that I want to train the model on devices with very restricted memory. At the moment I'm mostly interested in naive bayes and SVM. 

Another restriction I have is that I can't use floating-point numbers. I'll try to change the crate `math::num::RealNumber` to support [rust_decimal](https://docs.rs/rust_decimal/1.9.0/rust_decimal/).

**Comment by VolodymyrOrlov** [comment]
We definitely can add `partial_fit` to the Support Vector Classifier, since the [implementation we use](https://leon.bottou.org/projects/lasvm) is designed for incremental learning. 
Naive Bayes was implemented by Luis Moreno but it seems to me that we can add `partial_fit` to all NB variants we support, do you agree, @morenol?

@julama let me know if you want to work on this issue. In any case we can plan this story for next release (v0.3.0). Also, can I ask you to create a separate issue for `rust_decimal` (I want to keep these two stories separately)?

**Comment by morenol** [comment]
Yes, I think that we can add that support for the NB variants that we have implemented.

**Comment by julama** [comment]
> let me know if you want to work on this issue. In any case we can plan this story for next release (v0.3.0). Also, can I ask you to create a separate issue for `rust_decimal` (I want to keep these two stories separately)?

Yes, I'm interested to work on it. I just started learning rust but I will give it a try.

**Comment by Mec-iS** [comment]
moved to #279
